cloudinit/distros/photon.py

```diff
@@ -0,0 +1,380 @@
+#!/usr/bin/env python3
+# vi: ts=4 expandtab
+#
+# Copyright (C) 2021 VMware Inc.
+#
+# This file is part of cloud-init. See LICENSE file for license information.
+
+import os
+import fnmatch
+
+from cloudinit import util
+from cloudinit import subp
+from cloudinit import distros
+from cloudinit import helpers
+from cloudinit import log as logging
+from cloudinit.distros import net_util
+from cloudinit.settings import PER_INSTANCE
+from cloudinit.distros import rhel_util as rhutil
+from cloudinit.net.network_state import mask_to_net_prefix
+from cloudinit.distros.parsers.hostname import HostnameConf
+
+LOG = logging.getLogger(__name__)
+
+def network_file_name(dirname, dev_name):
+    network_file_pattern = '*' + dev_name + '.network*'
+    for node in os.listdir(dirname):
+        if fnmatch.fnmatch(node, network_file_pattern):
+            node_fullpath = os.path.join(dirname, node)
+            util.del_file(node_fullpath)
+            return dirname + node
+
+    return dirname + '10-' + dev_name + '.network'
+
+
+class Distro(distros.Distro):
+    hostname_conf_fn = '/etc/hostname'
+    locale_conf_fn = '/etc/default/locale'
+    network_conf_dir = '/etc/systemd/network/'
+    systemd_locale_conf_fn = '/etc/locale.conf'
+    resolve_conf_fn = '/etc/systemd/resolved.conf'
+
+    renderer_configs = {
+        'networkd': {
+            'resolv_conf_fn': resolve_conf_fn,
+            'network_conf_dir': network_conf_dir,
+            'network_fname_fptr': network_file_name,
+        }
+    }
+
+    # Should be fqdn if we can use it
+    prefer_fqdn = True
+
+    def __init__(self, name, cfg, paths):
+        distros.Distro.__init__(self, name, cfg, paths)
+        # This will be used to restrict certain
+        # calls from repeatly happening (when they
+        # should only happen say once per instance...)
+        self._runner = helpers.Runners(paths)
+        self.osfamily = 'photon'
+        self.init_cmd = ['systemctl']
+
+    def apply_locale(self, locale, out_fn=None):
+        if self.uses_systemd():
+            if not out_fn:
+                out_fn = self.systemd_locale_conf_fn
+        else:
+            if not out_fn:
+                out_fn = self.locale_conf_fn
+        locale_cfg = {
+            'LANG': locale,
+        }
+        rhutil.update_sysconfig_file(out_fn, locale_cfg)
```

Comment:
Is this correct? Are you using sysconfig?

Reply:
Yes, it's correct. I haven't enabled locate in cloud.cfg, we need glibc-i18n for this to work and we don't ship it by default. It just happens to be same as rhel way and I just need that utility function to update locale info.

However, I will remove the condition check for systemd, ours is systemd by default.